### ISSUE TITLE
asyncOnWrite operations should return a Promise

### ISSUE THREAD

**Issue by mnbroatch** [nmaggioni/Simple-JSONdb#8]
As it is, there doesn't seem to be a hook that lets the program know that writing is done, and as a result my database is randomly corrupted on program exit. Promise wrapper seems the most obvious way to handle this, but we need SOMETHING.

**Comment by nmaggioni** [comment]
If your concurrency model requires stronger guarantees I'd suggest you to move to a proper storage method instead of hardening this old toy JSON wrapper. Anyway, PRs are always welcome.

**Comment by iyobo** [comment]
It's not hard to roll your own json file disk storage honestly.
For starters, if you want to keep your data safe, all saving should happen to a different temp file and then, after success, replace main json file with temp file. 

This is how apps with file-based DBs like MSWord, etc do it to prevent corruption.
I do not know if this "toy" lib works that way.

**Comment by a-a-GiTHuB-a-a** [comment]
No, it doesn't...but honestly, I think adding Promises would be nice.

**Comment by a-a-GiTHuB-a-a** [comment]
I added a PR including this enhancement: <https://github.com/nmaggioni/Simple-JSONdb/pull/18>.
